### PR TITLE
Fallback to `CPUAccelerator` when `([], 0, "0")` are passed to the `devices` flag

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fallback to `CPUAccelerator` when `([], 0, "0")` are passed to the `devices` flag and to keep it consistent with the `gpus`, `tpu_cores` and `ipus` flag.

Current behavior with `gpus` and other device flags.

```python
## Doesn't fail and falls back to `CPUAccelerator`
Trainer(gpus=0)
```

Without the fix in this PR, the following used to fail
```python
Trainer(accelerator="xpu", devices=0)
```

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @akihironitta @rohitgr7 @justusschock

### CODE REVIEW

**Comment by rohitgr7** [comment]
```py
Trainer(gpus=0)
```
wouldn't it be better if this raises another error saying invalid values for `gpus` for 0 and []? this means the user has made a mistake while selecting the devices.

also 
```py
Trainer(gpus="0")
```
shouldn't train on GPU:0?

**Comment by kaushikb11** [comment]
>also

> Trainer(gpus="0")
> shouldn't train on GPU:0?

Nope, it falls back to `CPUAccelerator`. `Trainer(gpus="0,")` runs on GPU:0 
https://pytorch-lightning.readthedocs.io/en/latest/accelerators/gpu.html#select-gpu-devices

**Comment by four4fish** [comment]
Was this change introduced in https://github.com/PyTorchLightning/pytorch-lightning/pull/12410/ on purpose?

**Comment by rohitgr7** [comment]
> Was this change introduced in #12410 on purpose?

https://github.com/PyTorchLightning/pytorch-lightning/pull/12160#discussion_r816714587

**Comment by kaushikb11** [comment]
Closing this for #12633
